File: src/HelloOgre3D/sandbox/state/DeathState.cpp
#include "DeathState.h"
#include "object/AgentObject.h"
#include "GameDefine.h"

DeathState::DeathState(AgentObject* pAgent) : AgentState(pAgent)
{
	m_stateId = "DeathState";
}

DeathState::~DeathState()
{

}

void DeathState::OnEnter()
{
	m_pAgent->RequestState(SSTATE_DEAD);
}

void DeathState::OnLeave()
{

}

std::string DeathState::OnUpdate(float dt)
{
	if (m_pAgent->IsMoving())
	{
		m_pAgent->SlowMoving(2.0f);
	}
	else
	{
		if (m_pAgent->GetCurStateId() != SSTATE_DEAD)
		{
			m_pAgent->RequestState(SSTATE_DEAD);
		}
		else
		{
			m_pAgent->SetHealth(0.0f);
		}
	}

	return "";
}

File: bin/res/scripts/agent/states/DeathState.lua
-- DeathState --

--local m_pAgent = owner;
AgentState_OnEnter = function(agent)
    agent:RequestState(SSTATE_DEAD);
end

AgentState_OnLeave = function(agent)

end

AgentState_OnUpdate = function(agent, deltaTime)
    if agent:IsMoving() then
        agent:SlowMoving(2.0)
    else
        if agent:GetCurStateId() ~= SSTATE_DEAD then
            agent:RequestState(SSTATE_DEAD)
        else
            agent:SetHealth(0.0)
        end
    end

    return "";
end
